performance_test/src/communication_abstractions/resource_manager.cpp

```diff
@@ -239,4 +244,78 @@ void ResourceManager::connext_dds_micro_subscriber(
   }
 }
 #endif
+
+#ifdef PERFORMANCE_TEST_OPENDDS_ENABLED
+DDS::DomainParticipant_ptr
+ResourceManager::opendds_participant() const
+{
+  std::lock_guard<std::mutex> lock(m_global_mutex);
+  
+  if (CORBA::is_nil(m_opendds_participant)) {
+   DDS::DomainParticipantFactory_var dpf = TheParticipantFactory;
+
+    TransportConfig_rch config = TransportRegistry::instance()->create_config("ApexAiConfig");
+    TransportInst_rch inst = TransportRegistry::instance()->create_inst("rtps_tran","rtps_udp");
+ 
+    RtpsUdpInst_rch rui = static_rchandle_cast<RtpsUdpInst>(inst);
+    rui->handshake_timeout_ = 1;
+
+    config->instances_.push_back(inst);
+    TransportRegistry::instance()->global_config(config);
+
+    int domain = 0;
+    bool multicast = true;
+    unsigned int resend = 1;
+    std::string partition, governance, permissions;
+    int defaultSize = 0;
+
+    RtpsDiscovery_rch disc = make_rch<RtpsDiscovery>("RtpsDiscovery");
+    rui->use_multicast_ = true;
+    rui->local_address("127.0.0.1:");
+    rui->multicast_interface_="lo";
+    disc->sedp_multicast(true);
+
+    TheServiceParticipant->add_discovery(static_rchandle_cast<Discovery>(disc));
+    TheServiceParticipant->set_repo_domain(domain, disc->key());
```

Comment:
Is this domain the same as the domain ID? Should it use 0 or the domain id from the ExperimentConfiguration?

Reply:
corrected